### PR TITLE
Gradient clipping

### PR DESCRIPTION
**Pull request by Ranapop**
With this changes gradient clipping is added to the training.

Before these changes the model was at some point having a sudden increase in loss and a drop to 0 for the accuracy (from which it would not recover). A potential cause for such behaviour would be exploding gradients, which can be treated with gradient clipping.

After these changes, even if the model still has some spikes in the training, they are smaller and there's no drop to 0 in the accuracy.

### CODE REVIEW

**Review by marcvanzee — APPROVED** [review]
Hurray, gone at the exploding gradients!!
